src/main/webview/event/tasks/login.ts

```diff
@@ -0,0 +1,111 @@
+import { ILoginPage, ISendLoginEventData, LoginConnectionType, LoginProgressStatus, PageType } from 'jfrog-ide-webview';
+import { EventSender } from '../eventSender';
+import * as vscode from 'vscode';
+import { LogManager } from '../../../log/logManager';
+import { ConnectionManager, LoginStatus } from '../../../connect/connectionManager';
+import { ClientUtils } from 'jfrog-client-js';
+
+/**
+ * Represents a login task that handles the authentication process and communicates with the webview.
+ */
+export class LoginTask {
+    private updatedPageStatus: ILoginPage;
+    private platformUrl: string;
+    private artifactoryUrl: string;
+    private xrayUrl: string;
+    private username?: string;
+    private password?: string;
+    private accessToken?: string;
+
+    constructor(private send: EventSender, data: ISendLoginEventData, private connectionManager: ConnectionManager, private logManager: LogManager) {
+        this.updatedPageStatus = {
+            pageType: PageType.Login,
+            url: data.url || '',
+            status: LoginProgressStatus.Verifying,
+            connectionType: data.loginConnectionType
+        };
+        this.platformUrl = data.url !== undefined ? ClientUtils.addTrailingSlashIfMissing(data.url) : '';
+        this.artifactoryUrl = data.artifactoryUrl || this.platformUrl + 'artifactory';
+        this.xrayUrl = data.xrayUrl || this.platformUrl + 'xray';
+        this.username = data.username;
+        this.password = data.password;
+        this.accessToken = data.accessToken;
+    }
+
+    /**
+     * Executes the login task.
+     */
+    public async run() {
+        // Send initial page status to the webview
+        await this.send.loadPage(this.updatedPageStatus);
+
+        // Perform login and update page status
+        const requestStatus: LoginProgressStatus = await this.doLogin();
+        await this.send.loadPage({ ...this.updatedPageStatus, status: requestStatus });
+
+        // Connect to the IDE if login is successful
+        if (requestStatus === LoginProgressStatus.Success) {
+            await this.connectIde();
```

Comment:
Question - does it block the UI? If so, can we remove the "await"?

Reply:
Removed